policy/compiler/compiler.go

```diff
@@ -335,6 +335,41 @@ func (tc *templateCompiler) compileOpenAPISchema(dyn *model.DynValue,
 	if found {
 		schema.DefaultValue = tc.convertToSchemaType(elem.Ref.ID, elem.Ref.Value, schema)
 	}
+
+	tc.validateOpenAPISchema(dyn, schema)
+}
+
+func (tc *templateCompiler) validateOpenAPISchema(dyn *model.DynValue, schema *model.OpenAPISchema) {
+	m := tc.mapValue(dyn)
+	if len(m.Fields) > 0 && len(schema.Type) == 0 {
+		tc.reportErrorAtID(dyn.ID,
+			"invalid schema. fields not empty, type must be set.")
+	}
+	if len(schema.Properties) > 0 && schema.AdditionalProperties != nil {
```

Comment:
In this case, prefer to report the error at the `additionalProperties` location:

```
ap, _ := m.GetField("additionalProperties")
tc.reportErrorAtID(ap.Ref.ID, ...)
```

Reply:
done